src/aws/auth.rs

```diff
@@ -207,28 +232,20 @@ impl AwsAuthentication {
             }
             AwsAuthentication::Role {
                 assume_role,
-                load_timeout_secs,
```

Comment:
For other readers, this is covered by the new `credentials_cache` added that is wired up in `src/aws/mod.rs`.

Nit: not related to this PR but since it technically could apply the load timeout to all providers now since it happens at a different level, we could pull up `load_timeout_secs` out of the provider specific configuration.